src/cache/RrObjectCache.js

```diff
@@ -0,0 +1,44 @@
+import validateCacheSize from './util/validateCacheSize';
+import isStringOrNumber from './util/isStringOrNumber';
+
+export default class RrObjectCache {
+  constructor({cacheSize} = {}) {
+    validateCacheSize(cacheSize);
+    this.clear();
+    this._cacheSize = cacheSize;
+  }
+  set(key, selectorFn) {
+    if (this._cacheLength >= this._cacheSize) {
+      this._randomReplace(key, selectorFn);
+    } else {
+      this._cache[key] = selectorFn;
+      this._cacheKeys[this._cacheLength] = key;
+      this._cacheLength++;
+    }
+  }
+  get(key) {
+    return this._cache[key];
+  }
+  remove(key) {
+    const index = this._cacheKeys.indexOf(key); // O(1)
+    if (index > -1) {
+      delete this._cache[key];
+      this._cacheLength--;
+      this._cacheKeys[index] = this._cacheKeys[this._cacheLength];
+    }
+  }
+  clear() {
+    this._cache = {};
+    this._cacheKeys = [];
```

Comment:
A bit pedantic comment: `_cacheKeys` property is named `_cacheOrdering` in the existing cache implementations.

I think `_cacheKeys` is a better name but, for the sake of this PR, would you mind renaming `_cacheKeys` as `_cacheOrdering`? 🙏

Reply:
Sure, of course. I originally had it as `_cacheOrdering` but changed it to `_cacheKeys` because the order isn't strictly "relevant" to the eviction algorithm. However, the order is certainly necessary for the eviction.